@xen-orchestra/backups/_runners/VmsRemote.mjs

```diff
@@ -77,16 +83,38 @@ export const VmsRemote = class RemoteVmsBackupRunner extends Abstract {
             throttleStream,
             vmUuid,
           }
-          let vmBackup
-          if (job.mode === 'delta') {
-            vmBackup = new IncrementalRemote(opts)
-          } else if (job.mode === 'full') {
-            vmBackup = new FullRemote(opts)
-          } else {
-            throw new Error(`Job mode ${job.mode} not implemented for mirror backup`)
-          }
 
-          return runTask(taskStart, () => vmBackup.run())
+          return runTask(taskStart, () => {
+            return pRetry(
+              () => {
+                let vmBackup
+                switch (job.mode) {
+                  case 'delta':
+                    vmBackup = new IncrementalRemote(opts)
+                    break
+                  case 'full':
+                    vmBackup = new FullRemote(opts)
+                    break
+                  default:
+                    throw notImplementedJobModeError
+                }
+
+                return vmBackup.run()
+              },
+              {
+                delays: fibonacci()
+                  .take(allSettings[''].nRetriesVmBackupFailures ?? 0)
```

Comment:
this will use only the setting set globally, without the { ...settings, ...allSettings[vmUuid] } will contains the complete settings with default values correclty set . This can be tricky since the default settings can be overwritten by schedule / VM

You can add `nRetriesVmBackupFailures` in the `DEFAULT_REMOTE_VM_SETTINGS` to ensure the value is initialized